packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -186,7 +187,17 @@ export abstract class SharedObject<TEvent extends ISharedObjectEvents = ISharedO
     /**
      * {@inheritDoc (ISharedObject:interface).snapshot}
      */
-    public abstract snapshot(): ITree;
+    public snapshot(): ITree {
+        const serializer = new SnapshotSerializer(this.runtime.IFluidHandleContext);
+        return this.snapshotCore(serializer);
```

Comment:
I am not using the serialized routes tracked by the SnapshotSerializer for now. This will happen in a follow up PR where GC will use this information.